src/commonMain/kotlin/app/cash/turbine/flow.kt

```diff
@@ -204,27 +204,15 @@ private fun <T> testInInternal(flow: Flow<T>, timeout: Duration?, scope: Corouti
   return turbine
 }
 
-private fun <T> Flow<T>.collectTurbineIn(scope: CoroutineScope, timeout: Duration?, name: String?): Turbine<T> {
-  lateinit var channel: Channel<T>
-
+private fun <T> Flow<T>.collectTurbineIn(scope: CoroutineScope, timeout: Duration?, name: String?): ReceiveTurbine<T> {
   // Use test-specific unconfined if test scheduler is in use to inherit its virtual time.
   @OptIn(ExperimentalCoroutinesApi::class) // UnconfinedTestDispatcher is still experimental.
   val unconfined = scope.coroutineContext[TestCoroutineScheduler]
     ?.let(::UnconfinedTestDispatcher)
     ?: Unconfined
 
-  val job = scope.launch(unconfined, start = UNDISPATCHED) {
-    channel = collectIntoChannel(this)
-  }
```

Comment:
Now that tests aren't using `collectIntoChannel`, we can inline it and remove this pointless double launch.